### PR TITLE
Refactor interactive tests

### PR DESCRIPTION
**Pull request by ssi91**
This is an attempt to cover the issue https://github.com/ethereum/solidity/issues/6432

Not sure whether I've found all cases that could be moved to a common implementation though

### CODE REVIEW

File: test/TestCase.h [hunk 1]
```diff
@@ -102,7 +102,22 @@ class TestCase
 	bool m_shouldRun = true;
 };
 
-class EVMVersionRestrictedTestCase: public TestCase
+class CommonBaseTestCase: public TestCase
```
Comment (hrkrshnn) [inline test/TestCase.h]: Is there a reason you went with a new base class instead of implementing these additional functions in `TestCase`?

**Review by hrkrshnn — CHANGES_REQUESTED** [review]
Looks good to me. Could you please squash the commits?

File: test/TestCase.h (hunk 1 above)
Comment (ssi91) [reply test/TestCase.h]: At the beginning I thought it should be kind of mix-in class, but letter I've realised it's a base for all `TestCase` subclasses :)
@hrkrshnn Do you think worth to move the common implementation to the base `TestCase`?

Comment (hrkrshnn) [reply test/TestCase.h]: @ssi91 If every child of `TestClass` needs this, then I would prefer implementing these functions in `TestClass`.

Comment (ssi91) [reply test/TestCase.h]: well, I've done this

**Comment by sw7240614** [comment]
Thank you

On Friday, May 29, 2020, ssi91 <notifications@github.com> wrote:
> @ssi91 commented on this pull request.
>
> ________________________________
>
> In test/TestCase.h:
>
>> @@ -102,7 +102,22 @@ class TestCase
> bool m_shouldRun = true;
> };
>
> -class EVMVersionRestrictedTestCase: public TestCase
> +class CommonBaseTestCase: public TestCase
>
> well, I've done this
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.<
https://ci5.googleusercontent.com/proxy/GoHNAkoi43sgCfCvGxiIpqePz-soN6IhCndXEEd3fY8GpaHNDO7IO2ngDM0TV866Vjt6EN4ocar44ex1TqzCAQQaD-NFppukdcmj5l7Xb20KqK6rU2RzhNd0SEGlhdM4bvV1529kM9wmLMlorY-bwWGapLPVedk9EnInKej387afrUjVuuZDbdGgTQuDh9p_b_yQcrJbpMtYWlhFHA20gY8tPeGv3YLji1SJluJRUJG9vgI-A3hT=s0-d-e1-ft#https://github.com/notifications/beacon/APC5IPZGULWMOL47BYYHCH3RUAG6BA5CNFSM4NNL2TVKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGODEN3I3I.gif
>

-- 
null

**Review by hrkrshnn — DISMISSED** [review]
LGTM.

File: test/TestCase.cpp [hunk 2]
```diff
@@ -18,11 +18,12 @@
 #include <test/Common.h>
 #include <test/TestCase.h>
 
-#include <boost/algorithm/string/predicate.hpp>
 #include <boost/algorithm/string.hpp>
+#include <boost/algorithm/string/predicate.hpp>
+#include <libsolutil/AnsiColorized.h>
```
Comment (mijovic) [inline test/TestCase.cpp]: You need to move this line upper, for more info look here:
https://github.com/ethereum/solidity/blob/develop/CODING_STYLE.md#12-include-headers

**Review by mijovic — COMMENTED** [review]
Except the include order issue looks good

File: test/TestCase.cpp (hunk 2 above)
Comment (ssi91) [reply test/TestCase.cpp]: @mijovic Ok, It's done :)

**Review by ekpyron — APPROVED** [review]

**Review by chriseth — APPROVED** [review]

**Pull request merged**